test/mock-interceptor.js

```diff
@@ -53,6 +53,23 @@ test('MockInterceptor - reply callback', t => {
     t.throws(() => mockInterceptor.reply(), new InvalidArgumentError('statusCode must be defined'))
     t.throws(() => mockInterceptor.reply(200, () => {}, 'hello'), new InvalidArgumentError('responseOptions must be an object'))
   })
+})
+
+test('MockInterceptor - reply options callback', t => {
+  t.plan(1)
+
+  t.test('should return MockScope', t => {
+    t.plan(1)
+    const mockInterceptor = new MockInterceptor({
+      path: '',
+      method: ''
+    }, [])
+    const result = mockInterceptor.reply(() => ({
```

Comment:
Can you add a few assertions on the parameters of this function?

Reply:
So I had a question about this. Since callbacks are late-binding, their validation errors for `status`, `data` and `responseOptions` get thrown later in the actual fetch operation. This leads these validation errors to show up as `fetch failed` errors. Would there be any way to expose the true error during the fetch operation instead of just getting the standard `fetch failed` error?

Reply:
Oh actually never mind, it's attached the the `cause` of the fetched failed error